### PR TITLE
Clean up channel options

### PR DESCRIPTION
**Pull request by joshlory**
The only usage of `isChannelBacked` is in project.js when deciding whether a project is editable. I _think_ once we get to this decision point, the presence of `appOptions.channel` is enough to tell that a project is channel-backed.

A bit more discussion here: https://codedotorg.slack.com/archives/C0T0PNR0D/p1550689911000600

### CODE REVIEW

**Comment by joshlory** [comment]
The `isChannelBacked` view option was added in commit https://github.com/code-dot-org/code-dot-org/commit/27fcb80416c88aabe88faa11924072215d9780e8.

File: dashboard/app/helpers/levels_helper.rb [hunk 1]
```diff
@@ -95,7 +95,7 @@ def get_channel_for(level, user = nil)
       )
     end
 
-    channel_token.try :channel
+    channel_token.channel if channel_token
```
Comment (davidsbailey) [inline dashboard/app/helpers/levels_helper.rb]: can also use ruby's new'ish `channel_token&.channel` syntax. any of these options are fine though

File: apps/src/code-studio/initApp/project.js [hunk 2]
```diff
@@ -1343,7 +1343,7 @@ var projects = (module.exports = {
         isEditing = true;
         deferred.resolve();
       }
-    } else if (appOptions.isChannelBacked) {
+    } else if (appOptions.channel) {
```
Comment (davidsbailey) [inline apps/src/code-studio/initApp/project.js:1346]: logic update looks good!

**Review by davidsbailey — APPROVED** [review]
Nice cleanup!

File: dashboard/app/helpers/levels_helper.rb (hunk 1 above)
Comment (joshlory) [reply dashboard/app/helpers/levels_helper.rb]: Ah good call! I can never remember which side of the expression is allowed to be nil with the save-navigation operator.

For my own reference...

```ruby
[1] pry(main)> nil&.foo
=> nil
[2] pry(main)> 5&.foo
NoMethodError: undefined method `foo' for 5:Integer
Did you mean?  floor
from (pry):44:in `__pry__'
```

**Pull request merged**